Documentation updates for v3.1 (#3475)
geopm/geopm: docs/source/overview.rst

----

|:computer:| Install GEOPM
--------------------------

Pre-built binaries of the GEOPM Service and Runtime are available for download
using the openSUSE Build Service for RPM-based Linux distributions and through
Launchpad for Debian-based distributions.  For information on how to configure
those repositories with your system package manager or directly download the
binaries see: :doc:`install`

Spack
^^^^^

.. note::

    Recipes for the v3.1 release of GEOPM are currently work-in-progress.

For users that leverage `Spack <https://spack.io/>`_ to distribute software,
recipes to build the `geopm-service
<https://github.com/spack/spack/blob/v0.22.0/var/spack/repos/builtin/packages/geopm-service/package.py>`_
and `geopm-runtime
<https://github.com/spack/spack/blob/v0.22.0/var/spack/repos/builtin/packages/geopm-runtime/package.py>`_
have been included in their `v0.22.0 release
<https://github.com/spack/spack/tree/v0.22.0>`_.  These recipes currently allow
for building both the v3.0.1 release of GEOPM and our main development branch.

> For deploying GEOPM's layers to a compute image, a typical configration would

Reviewer: ```suggestion
For deploying GEOPM's layers to a compute image, a typical configuration would
```

> be to have a system install of the service RPMs baked into the compute image,

Reviewer 2: ```suggestion
be to have a system install of the service RPMs baked into the compute image
```

and use spack to install ``geopm-runtime``.  This is required as the GEOPM
service will be launched via systemd, and thus must run against the system
installed Python runtime.

For GEOPM v3.0.1, system install ``geopm-service``, ``libgeopmd2``, and ``python3-geopmdpy``

For GEOPM v3.1, system install ``geopm-service``, ``libgeopmd2``, ``python3-geopmdpy``, and ``python3-geopmdpy-doc``

In order to build with spack this way, ``geopm-service`` must be configured as an
external package in ``~/.spack/packages.yaml``:

> .. code-block:: bash

Reviewer: bash -> yaml?

100 W.

Video Demo: Using ``geopmwrite``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.. raw:: html

    <video src="images/geopmwrite.webm" type="video/webm" controls="controls" muted="muted" class="d-block rounded-bottom-2 border-top width-fit" style="max-height:640px; min-height: 200px"></video>

----

|:straight_ruler:| Measure Performance
--------------------------------------

The GEOPM Runtime offers capabilities for collecting telemetry throughout an

> application's execution. If you're aiming to measure a particular segment of an

Reviewer 2: ```suggestion
application's execution. If you want to measure a particular segment of an
```

Reviewer 2: Colloquialism might be difficult for non-native English speakers.
